bazel/cert.bzl

```diff
@@ -0,0 +1,130 @@
+"""
+This module contains functions to generate a simple CA
+"""
+
+# buildifier: disable=function-docstring-args
+def _redpanda_private_key(name, certificate):
+    private_key = certificate + ".key"
+
+    pr_key_gen = name + "_key_gen"
+    native.genrule(
+        name = pr_key_gen,
+        srcs = [],
+        outs = [private_key],
+        cmd = "$(execpath @openssl//:openssl_exe) ecparam " +
+              "-name prime256v1 " +
+              "-genkey " +
+              "-noout " +
+              "-out \"$@\"",
+        tools = [
+            "@openssl//:openssl_exe",
+        ],
+    )
+
+    return pr_key_gen
+
+def redpanda_selfsigned_cert(name, certificate, common_name):
+    """
+    Generate a Redpanda self-signed certificate.
+
+    Args:
+      name: name of the target
+      certificate: name to use for output files (crt, key, and csr)
+      common_name: the CN to use when setting the subject name
+    """
+
+    cert = certificate + ".crt"
+    subj = "/C=US/ST=California/L=San Francisco/O=Redpanda Data/OU=Core/CN=" + common_name
+
+    pr_key_gen = _redpanda_private_key(name, certificate)
+
+    crt_gen = name + "_crt_gen"
+    native.genrule(
+        name = crt_gen,
+        srcs = [
+            pr_key_gen,
+        ],
+        outs = [cert],
+        cmd = "$(execpath @openssl//:openssl_exe) req " +
+              "-new -x509 -sha256 " +
+              "-key $(SRCS) " +
+              "-out \"$@\" " +
+              "-subj \"{}\" ".format(subj) +
+              "-addext \"subjectAltName = IP:127.0.0.1\"",
+        tools = [
+            "@openssl//:openssl_exe",
+        ],
+    )
+
+    native.filegroup(
+        name = name,
+        srcs = [pr_key_gen, crt_gen],
+        visibility = [
+            "//visibility:public",
```

Comment:
same as the other, this should be passed in by the user and I also think we can be more strict here